addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -390,6 +492,68 @@ function ThreadFactory({ Entity }) {
             }
         }
 
+        /**
+         * Refresh the is typing status of the current partner.
+         */
+        refreshCurrentPartnerIsTyping() {
+            this._currentPartnerInactiveTypingTimer.reset();
+        }
+
+        /**
+         * Called to refresh a registered other member partner that is typing
+         * something.
+         *
+         * @param {mail.messaging.entity.Partner} partner
+         */
+        refreshOtherMemberTypingMember(partner) {
+            this._otherMembersLongTypingTimers.get(partner).reset();
+        }
+
+        /**
+         * Called when current partner is inserting some input in composer.
+         * Useful to notify current partner is currently typing something in the
+         * composer of this thread to all other members.
+         */
+        async registerCurrentPartnerIsTyping() {
+            // Handling of typing timers.
+            this._currentPartnerInactiveTypingTimer.start();
+            this._currentPartnerLongTypingTimer.start();
+            // Manage typing member relation.
+            const currentPartner = this.env.messaging.currentPartner;
+            const newOrderedTypingMemberLocalIds = this.orderedTypingMemberLocalIds
+                .filter(localId => localId !== currentPartner.localId);
+            newOrderedTypingMemberLocalIds.push(currentPartner.localId);
+            this.update({
+                typingMembers: [['link', currentPartner]],
+                orderedTypingMemberLocalIds: newOrderedTypingMemberLocalIds,
+            });
```

Comment:
The current partner is not supposed to see himself as typing, why update `orderedTypingMemberLocalIds` in this case?

Reply:
Currently UI omit current partner in typing, but this is decision that may change in future (not that I foresee it will definitely be the case, just think that "and why not display it?" could somewhat make sense from a functional PoV), so that's why I wouldn't enforce typing members = all typing members except current partner (and that's the reason I introduced a computed field for other typing members that omit current partner).

Reply:
ok